tools/Performance/DynamoPerformanceTests/ResultsComparer.cs

```diff
@@ -177,48 +200,50 @@ internal List<string[]> GetComparisonData()
             }
         }
         
-        private List<BenchmarkComparison> Comparisons;
-
-        private string BasePath;
-        private string NewPath;
-
         /// <summary>
         /// Creates a results comparer to compare results at basePath and newPath
         /// </summary>
         /// <param name="basePath"></param>
         /// <param name="newPath"></param>
-        public ResultsComparer(string basePath, string newPath)
+        public ResultsComparer(string basePath, string newPath, string saveCSVPath = null)
         {
-            BasePath = basePath;
-            NewPath = newPath;
+            // Create BenchmarkComparison objects
+            var comparisons = CreateComparisons(basePath, newPath);
+            
+            // Log comparisons to console
+            LogResults(comparisons);
 
-            CreateComparisons();
-            LogResults();
+            // Save comparisons to csv if path provided
+            if (saveCSVPath != null && saveCSVPath != string.Empty)
+            {
+                WriteResultsToCSV(comparisons, saveCSVPath);
+            }
         }
 
         /// <summary>
         /// Compare the two results files and build comparison data
         /// </summary>
-        private void CreateComparisons()
+        private IEnumerable<BenchmarkComparison> CreateComparisons(string basePath, string newPath)
```

Comment:
Minor comment - it usually is okay to use concrete types like `List` as opposed to interfaces in private methods since we have a control over them. Concrete types are generally faster.